editor/components/document-outline/item.js

```diff
@@ -28,6 +30,15 @@ const TableOfContentsItem = ( {
 			onClick={ onClick }
 		>
 			<span className="document-outline__emdash" aria-hidden="true"></span>
+			{
+				// path is an array of nodes that are ancestors of the heading starting in the top level node.
+				// This mapping renders each ancestor to make it easier for the user to know where the headings are nested.
+				path.map( ( { name }, index ) => (
+					<strong key={ index } className="document-outline__level">
+						{ getBlockType( name ).title }
```

Comment:
Minor: There is a `BlockTitle` component now, which could either be accessed by its connected form (pass UID), or its raw form taking `name` as a prop.

https://github.com/WordPress/gutenberg/tree/master/editor/components/block-title